### PR TITLE
Update before posting

### PR DESCRIPTION
**Pull request by xiaoruiDong**
This PR includes necessary changes to allow training with frozen weights and helper scripts for inference.

### CODE REVIEW

File: paper_results/scripts/predict_DMPNN1.sh [hunk 1]
```diff
@@ -0,0 +1,104 @@
+#!/bin/bash -l
+
+####### Input Parameters ########
+
+# Path to the paper_results folder
+# The script set this path automatically by default
+# However, if runing into issues, you may manually set this path
```
Comment (kspieks) [inline paper_results/scripts/predict_DMPNN1.sh]: Can you please change "runing" to "running"?

File: paper_results/scripts/predict_DMPNN1.sh [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+paper_results_dir=$(dirname $(dirname "$(realpath "$0")"))
+# paper_results_dir="/path/to/chemprop/paper_results"
+
+# Input file
+# A csv file containing reaction SMILES
+test_path=example_reactions.csv
+rxn_column="rxn_smiles"s
```
Comment (kspieks) [inline paper_results/scripts/predict_DMPNN1.sh]: Is the `s` supposed to be here?

File: paper_results/scripts/predict_DMPNN1.sh [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+num_workers=2
+
+# Batch size used during inference
+batch_size=32
+
+####### End of Input Parameters ########
+
+# Print configuration
+echo "##########################################################################"
+echo "# Predict Gibbs free energy of activation and of reaction using D-MPNN-1 #"
```
Comment (kspieks) [inline paper_results/scripts/predict_DMPNN1.sh]: Can we be more specific with this print statement about which ΔG is being predicted? e.g., specify that it's the Gibbs free energy of activation and of reaction in solution (water)?

File: paper_results/scripts/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+### Introduction for included scripts
+- `download_dataset.sh` can be used to automatically download datasets from Zenodo to `/paper_results/datasets`.
+- `predict_DMPNN1.sh` can be used to conduct inference using D-MPNN-1 model. You may open the file and modify the arguments according to your need and setup.
+- `predict_DMPNN2.sh` can be used to conduct inference using D-MPNN-2 model. You may open the file and modify the arguments according to your need and setup.
```
Comment (kspieks) [inline paper_results/scripts/README.md]: Can we add a note here that recommends users only make predictions using one of the 40 solvents from the paper? I assume the model will perform poorly if users provide a different solvent so we should just give them a heads up about that

File: paper_results/scripts/merge_data.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    df_gas = pd.read_csv(preds_path + ".gas")
+    df_solv = pd.read_csv(preds_path + ".solv")
+
+    # Combine prediction
+    df_gas = df_gas.rename(columns={rxn_column: f'{rxn_column}_gas'})
+    df_pred = df_solv.merge(df_gas, how='left', left_on=rxn_column, right_on=f'{rxn_column}_gas')
+    df_pred = pd.concat([df_solv, df_gas], axis=1)
+    df_pred = df_pred.drop(columns=[f'{rxn_column}_gas'])
+
+    # Computat solvated barrier heights
```
Comment (kspieks) [inline paper_results/scripts/merge_data.py]: Should this say "Compute solvated barrier heights"?

**Review by kspieks — CHANGES_REQUESTED** [review]
Thanks Xiaorui! This PR looks great. Thanks for creating organized scripts with excellent documentation and comments. I had a few minor questions/comments but we can definitely merge soon

File: paper_results/scripts/merge_data.py (hunk 5 above)
Comment (xiaoruiDong) [reply paper_results/scripts/merge_data.py]: Thanks

**Comment by xiaoruiDong** [comment]
@kspieks Thank you for reviewing this PR! Good catches!  Typos are fixed, and instructions are updated.

**Comment by kspieks** [comment]
Thanks Xiaorui. LGTM

**Pull request merged**